x-pack/test/upgrade/apps/canvas/canvas_smoke_tests.ts

```diff
@@ -21,49 +21,35 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
     ];
 
     const canvasTests = [
-      {
-        name: 'flights',
-        id: 'workpad-a474e74b-aedc-47c3-894a-db77e62c41e0/page/1',
-        altId: '',
-        numElements: 35,
-      },
-      {
-        name: 'logs',
-        id: 'workpad-5563cc40-5760-4afe-bf33-9da72fac53b7/page/1',
-        altId: 'workpad-ad72a4e9-b422-480c-be6d-a64a0b79541d',
-        numElements: 57,
-      },
-      {
-        name: 'ecommerce',
-        id: 'workpad-e08b9bdb-ec14-4339-94c4-063bddfd610e/page/1',
-        altId: '',
-        numElements: 16,
-      },
-      {
-        name: 'ecommerce',
-        id: 'workpad-e08b9bdb-ec14-4339-94c4-063bddfd610e/page/2',
-        altId: '',
-        numElements: 9,
-      },
+      { name: 'flights', page: 1, numElements: 35 },
+      { name: 'logs', page: 1, numElements: 57 },
+      { name: 'ecommerce', page: 1, numElements: 16 },
+      { name: 'ecommerce', page: 2, numElements: 9 },
     ];
 
     spaces.forEach(({ space, basePath }) => {
-      canvasTests.forEach(({ name, id, altId, numElements }) => {
-        describe('space ' + space + ' name ' + name, () => {
-          beforeEach(async () => {
-            await PageObjects.common.navigateToActualUrl('canvas', 'workpad/' + id, {
-              basePath,
-            });
+      describe('space ' + space, () => {
+        beforeEach(async () => {
+          await PageObjects.common.navigateToActualUrl('home', '/tutorial_directory/sampleData', {
+            basePath,
+          });
+          await PageObjects.header.waitUntilLoadingHasFinished();
+        });
+        canvasTests.forEach(({ name, numElements, page }) => {
+          it('renders elements on workpad ' + name + ' page ' + page, async () => {
+            await PageObjects.home.launchSampleCanvas(name);
             await PageObjects.header.waitUntilLoadingHasFinished();
-            const url = await browser.getCurrentUrl();
-            if (!url.includes(id) && altId.length > 0) {
-              await PageObjects.common.navigateToActualUrl('canvas', 'workpad/' + altId, {
-                basePath,
-              });
+            const currentUrl = await browser.getCurrentUrl();
+            const [, hash] = currentUrl.split('#/');
+            if (hash.length === 0) {
+              throw new Error('Did not launch canvas sample data for ' + name);
+            }
+            if (name === 'ecommerce') {
```

Comment:
Nit: Is there a better way to do this? perhaps by looping through the pages as well? That way if we ever change the number of pages in a demo data canvas, it doesn't break this test?

Reply:
@ThomThomson thanks for the feedback, when we add a page, there is still a manual step of adding what the verification is needed for that page, I can add a loop at some point if those updates are made, right now we have only one canvas type: ecommerce, that has two pages, the others are just one page, which is why it is done this way for now.